lamindb/_from_values.py

```diff
@@ -110,7 +109,7 @@ def get_or_create_records(
         #         logger.debug(f"added default feature '{feature_name}'")
         return RecordList(records)
     finally:
-        settings.creation.search_names = creation_search_names
+        settings.creation.search_names = True  # need to refactor the whole function because can no longer use settings; not thread safe
```

Comment:
@sunnyosun, we gotta eliminate the use of settings in `from_values()` entirely.

This is a big refactor in which we can also look into the dangers of creating duplicated records and similar.

I don't want to make this refactor here in this PR as it'd get too complicated (and I'm not so familiar with the `_from_values.py` file.

Reply:
Slack thread: https://laminlabs.slack.com/archives/C04FPE8V01W/p1739365084531499